src/Atmos/Model/AtmosModel.jl

```diff
@@ -716,4 +718,107 @@ function init_state_conservative!(
 )
     m.init_state_conservative(m, state, aux, coords, t, args...)
 end
+
+struct RoeNumericalFlux <: NumericalFluxFirstOrder end
+function numerical_flux_first_order!(
+    numerical_flux::RoeNumericalFlux,
+    balance_law::AtmosModel,
+    fluxᵀn::Vars{S},
+    normal_vector::SVector,
+    state_conservative⁻::Vars{S},
+    state_auxiliary⁻::Vars{A},
+    state_conservative⁺::Vars{S},
+    state_auxiliary⁺::Vars{A},
+    t,
+    direction,
+) where {S, A}
+
+    numerical_flux_first_order!(
+        CentralNumericalFluxFirstOrder(),
+        balance_law,
+        fluxᵀn,
+        normal_vector,
+        state_conservative⁻,
+        state_auxiliary⁻,
+        state_conservative⁺,
+        state_auxiliary⁺,
+        t,
+        direction,
+    )
+
+    FT = eltype(fluxᵀn)
+    param_set = balance_law.param_set
+    _cv_d::FT = cv_d(param_set)
+    _T_0::FT = T_0(param_set)
+    γ::FT = cp_d(param_set) / cv_d(param_set)
+
+    Φ = gravitational_potential(balance_law, state_auxiliary⁻)
+
+    ρ⁻ = state_conservative⁻.ρ
+    ρu⁻ = state_conservative⁻.ρu
+    ρe⁻ = state_conservative⁻.ρe
+
+    u⁻ = ρu⁻ / ρ⁻
+    p⁻ = pressure(
+        balance_law,
+        balance_law.moisture,
+        state_conservative⁻,
+        state_auxiliary⁻,
+    )
+    h⁻ = (ρe⁻ + p⁻) / ρ⁻
+
+    ρ⁺ = state_conservative⁺.ρ
+    ρu⁺ = state_conservative⁺.ρu
+    ρe⁺ = state_conservative⁺.ρe
+
+    u⁺ = ρu⁺ / ρ⁺
+    p⁺ = pressure(
+        balance_law,
+        balance_law.moisture,
+        state_conservative⁺,
+        state_auxiliary⁺,
+    )
+    h⁺ = (ρe⁺ + p⁺) / ρ⁺
+
+    ũ = (sqrt(ρ⁻) * u⁻ + sqrt(ρ⁺) * u⁺) / (sqrt(ρ⁻) + sqrt(ρ⁺))
+    h̃ = (sqrt(ρ⁻) * h⁻ + sqrt(ρ⁺) * h⁺) / (sqrt(ρ⁻) + sqrt(ρ⁺))
+    c̃ = sqrt((γ - 1) * (h̃ - ũ' * ũ / 2 - Φ + _cv_d * _T_0))
```

Comment:
Use ` c̃ = soundspeed_air(...)` from `Thermodynamics` here for this to be correct in the moist case.

Reply:
(Note that the error here otherwise is O(1) because the enthalpy has O(1) contributions from latent enthalpies.)

Reply:
To improve the low-Mach number behavior of the fluxes (which we need), you may want to first compute 
- `c = soundspeed_air(...)`
and then take 
- `c̃ =c * max(min(M, 1), M_cut)` where `M = abs(u)/c` and `M_cut` is a cutoff Mach number parameter (default value `M_cut = 0`.

See https://doi.org/10.1016/j.jcp.2008.01.037 (their Eq. (18) uses a cutoff function that goes smoothly to 1, and you could use that too.) This gives better results at low Mach numbers, which we have in LES. See https://arxiv.org/pdf/2004.01627.pdf for numerical examples.

Reply:
> Use ` c̃ = soundspeed_air(...)` from `Thermodynamics` here for this to be correct in the moist case. (@charleskawczynski can help if there are questions about how to access this.)

@charleskawczynski I wonder if you have already looked at this? (I am trying to implement moist Roe flux, and it seems there is no straightforward way to get soundspeed_air from enthalpy and velocity?)

Reply:
Why don't you do something like this:

- Define function `Roe_average(A⁻, A⁺, ρ⁻, ρ⁺) = ...`
- Compute `ũ = Roe_average(...), `\tilde e = Roe_average(e⁻, e⁺, ...)`, `\tilde T  = Roe_average(T⁻, T⁺, ...)`, `\tilde q = Roe_average...`
- Define thermodynamic state on basis of Roe averages
- Compute `c  = soundspeed_air(...)` [as function of thermodynamic state]
- Compute `h̃ = total_specific_enthalpy(\tilde e, Rm(\tilde q), tilde T)`

Reply:
Yes, that's the plan actually. Thanks!

Reply:
Although the `\tilde h` you get from this would not be the same as the Roe average of h?

Reply:
Yes, not exactly because of the nonlinearity in the moist gas constant. But this is a thermodynamically consistent way of defining the Roe averaged enthalpy through a Legendre transformation of thermodynamic states from the energy. My understanding is that it holds to the same degree of accuracy as the Roe averages in general.

Reply:
Good to know, thank you!

Reply:
What should be the density used for defining the thermodynamic state on basis of Roe averages? (sorry if it's a too basic question...)

Reply:
Yes, maybe better to avoid the thermodynamic state. 

If you want, you can also avoid the issue of the different `p/\rho` and `R_m T` Roe averages by simply Roe averaging `R_m T = p/\rho`, and then using that to compute the speed of sound (e.g., add another overloaded method that takes `R_m T` as input to compute `c`). However, practically the small difference between the two averages probably makes no difference.